app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt

```diff
@@ -76,14 +76,19 @@ class RadioSearchEngineListPreference @JvmOverloads constructor(
             ViewGroup.LayoutParams.WRAP_CONTENT,
         )
 
+        val isLastGeneralOrCustomSearchEngine = state.searchEngines.filter {
+            it.isGeneral ||
+                it.type == SearchEngine.Type.CUSTOM
+        }.size == 1
         state.searchEngines.filter { engine ->
             engine.type != SearchEngine.Type.APPLICATION
         }.forEach { engine ->
+            val isGeneralOrCustomSearchEngine = engine.isGeneral || engine.type == SearchEngine.Type.CUSTOM
             val searchEngineView = makeButtonFromSearchEngine(
                 engine = engine,
                 layoutInflater = layoutInflater,
                 res = context.resources,
-                allowDeletion = state.searchEngines.size > 1,
+                allowDeletion = !(isGeneralOrCustomSearchEngine && isLastGeneralOrCustomSearchEngine),
```

Comment:
This is needed in order to not allow the last general/custom search engine to be removed, therefore leading to not having one to set as default search engine. Open to feedback on how else this could be handled.